eventstore/test_views.py

```diff
@@ -387,6 +395,284 @@ def test_postbirth_other_passport_origin(self):
         self.assertEqual(registration.passport_country, "other")
 
 
+class MessagesViewSetTests(APITestCase):
+    url = reverse("messages-list")
+
+    def generate_hmac_signature(self, data, key):
+        data = JSONRenderer().render(data)
+        h = hmac.new(key.encode(), data, sha256)
+        return base64.b64encode(h.digest()).decode()
+
+    def test_message_identification_required(self):
+        """
+        There must be identification of user to make the request
+        """
+        response = self.client.post(self.url)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_message_authentication_required(self):
+        """
+        The user needs the `add _message` permission
+        in order to make the request
+
+        """
+        user = get_user_model().objects.create_user("test")
+        self.client.force_authenticate(user)
+        data = {"random": "data"}
+        response = self.client.post(
+            self.url,
+            data,
+            format="json",
+            HTTP_X_TURN_HOOK_SIGNATURE=self.generate_hmac_signature(data, "REPLACEME"),
+        )
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_message_authentication_errors(self):
+        """
+        We expect to get auth errors when no user token is added to the querystring,
+        but when added the request is made
+        """
+
+        user = get_user_model().objects.create_user("test")
+        user.user_permissions.add(Permission.objects.get(codename="add_message"))
+        data = {"random": "data"}
+
+        response = self.client.post(
+            self.url,
+            data,
+            format="json",
+            HTTP_X_TURN_HOOK_SIGNATURE=self.generate_hmac_signature(data, "REPLACEME"),
+        )
```

Comment:
We're not asserting anything on this request. We can either remove it, or assert something useful about it (eg. that we get an auth error)

Reply:
forgot to add the line, thanks for pointing it out